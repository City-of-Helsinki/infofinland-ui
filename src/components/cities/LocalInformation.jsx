import Button from '@/components/Button'
import { IconMapMarker } from '@/components/Icons'
import ParseHtml from '@/components/ParseHtml'
import { useAtom } from 'jotai'
import { selectedCity, cityMenuVisibility } from '@/src/store'
import Block from '@/components/article/Block'
import cls from 'classnames'
import { IconExternalSite } from '@/components/Icons'

import { READMORE_CONTENT } from '@/components/article/ReadMoreBlock'
const DEMOHTML = `
<div>

<p>Vocational training is aimed at both young people and adults. You can apply for vocational training all year round. You can study for an upper secondary education or increase your skills in vocational training. Vocational training in Vaasa is organised by Vamia and YA – Vocational College of Ostrobothnia (Yrkesakademin i Österbotten (YA)).</p>
<p>YA and Vamia also organise preparatory training for basic vocational training, i.e. VALMA. The training lasts for a maximum of one academic year. YA’s VALMA training is in Swedish; Handledande utbildning för yrkesutbildning.</p>
</div>
`
const LocalInformation = () => {
  const [city] = useAtom(selectedCity)
  // eslint-disable-next-line no-unused-vars
  const [, setOpen] = useAtom(cityMenuVisibility)
  const openMenu = () => setOpen(true)

  return (
    <div className="mb-8">
      <Block className="bg-green-lighter md:rounded">
        <div className=" md:flex md:justify-around items-center py-6">
          {!city && (
            <h3 className="md:flex-grow text-h3 font-bold">
              <IconMapMarker className="h-9 me-2 md:me-4" />
              Local information
            </h3>
          )}
          {city && (
            <IconMapMarker
              className={cls('me-2 md:me-4 h-9 inline-block', {
                'mt-2': city,
              })}
            />
          )}
          {city && (
            <h3 className="flex-grow text-h3 font-bold">
              <span className="block text-action font-normal text-dark">
                Local information
              </span>
              {city}
            </h3>
          )}
          <div className="md:flex-none mt-6 md:mt-0 mb-2 md:mb-0">
            <Button onClick={openMenu}>
              {!city && 'Choose City'}
              {city && 'Change City'}
            </Button>
          </div>
        </div>
      </Block>
      {city && (
        <Block className="py-8 mb-4 bg-green-white">
          <ParseHtml html={DEMOHTML} />
          <LocalReadMore />
        </Block>
      )}
    </div>
  )
}

const LocalReadMore = ({ content = READMORE_CONTENT }) => {
  return (
    <div className="p-4 bg-white rounded">
      {content.map(({ siteUrl, siteName, pageUrl, pageName, languages }, i) => (
        <div
          className={cls({
            'mb-3 pb-3 border-b border-gray-hr': i + 1 < content.length,
          })}
          key={`${siteName}-${i}`}
        >
          <span
            className="flex items-center text-small"
            href={siteUrl}
            target="_blank"
            rel="noreferrer"
          >
            <IconExternalSite className="me-2" />
            {siteName}
          </span>

          <a
            rel="noreferrer"
            href={pageUrl}
            target="_blank"
            className="inline-block mb-4 text-body font-bold ifu-text-link"
          >
            {pageName}
          </a>

          <div className="flex flex-wrap leading-4 divide-link divide-s">
            {languages.map(({ url, text, lang }, k) => (
              <a
                title={pageName}
                rel="noreferrer"
                href={url}
                key={`link-${text}-${k}`}
                target="_blank"
                lang={lang}
                className={cls('text-small ifu-text-link pe-2', {
                  'ps-2': k > 0,
                })}
              >
                {text}
              </a>
            ))}
          </div>
        </div>
      ))}
    </div>
  )
}

export default LocalInformation
