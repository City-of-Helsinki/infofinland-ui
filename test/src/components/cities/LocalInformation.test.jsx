/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import LocalInformation from '@/components/cities/LocalInformation'
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))
useRouter.mockImplementation(() => ({
  locale: 'fi'
}))
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (i18nKey) => i18nKey,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('LocalInformation', () => {
  it('should render', () => {
    const { asFragment } = render(<LocalInformation cities={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
