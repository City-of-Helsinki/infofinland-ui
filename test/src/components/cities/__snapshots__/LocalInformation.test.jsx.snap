// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalInformation should render 1`] = `
<DocumentFragment>
  <div
    class="mb-8"
  >
    <div
      class="flex items-center h-14 lg:h-16 bg-green-lighter lg:rounded-t ifu-block--article"
    >
      <h3
        class="md:flex-grow text-body font-bold -translate-x-4"
      >
        <svg
          class="inline-block h-7 me-2"
          fill="none"
          height="42"
          viewBox="0 0 36 42"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 21C15.525 21 13.5 19.11 13.5 16.8C13.5 14.49 15.525 12.6 18 12.6C20.475 12.6 22.5 14.49 22.5 16.8C22.5 19.11 20.475 21 18 21ZM31.5 17.22C31.5 9.597 25.5375 4.2 18 4.2C10.4625 4.2 4.5 9.597 4.5 17.22C4.5 22.134 8.8875 28.644 18 36.414C27.1125 28.644 31.5 22.134 31.5 17.22ZM18 0C27.45 0 36 6.762 36 17.22C36 24.192 29.9925 32.445 18 42C6.0075 32.445 0 24.192 0 17.22C0 6.762 8.55 0 18 0Z"
            fill="black"
          />
        </svg>
        localInfo.title
      </h3>
    </div>
    <div
      class="py-8 mb-4 bg-green-white ifu-block--article"
    >
      <div
        class="flex"
      >
        <button
          class="inline-block flex-none text-body-large font-bold"
          id="ifu-localinfo__button"
        >
          <span
            class="underline"
          >
            localInfo.select
          </span>
          <svg
            class="inline-block w-3 h-3 fill-black ms-2"
            fill="none"
            height="6"
            viewBox="0 0 11 6"
            width="11"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.3311 1.83871L5.32299 6L5.00847 6L0.000408195 1.83871L1.04073 -8.81717e-08L5.17783 3.3629L9.31493 -9.64378e-09L10.3311 1.83871Z"
            />
          </svg>
        </button>
        <div
          class="flex-grow"
        />
      </div>
      <p
        class="mt-2"
      >
        <label
          for="ifu-localinfo__button"
        >
          localInfo.help
        </label>
      </p>
      <ul
        class="overflow-hidden absolute w-0 h-0 ifu-localinfo__seo"
      />
    </div>
  </div>
</DocumentFragment>
`;
